gem/gem.py

```diff
@@ -876,7 +876,9 @@ def strides_of(shape):
 
 def decompose_variable_view(expression):
     """Extract ComponentTensor + FlexiblyIndexed view onto a variable."""
-    if isinstance(expression, Variable):
+    if (isinstance(expression, Variable) or
+        (isinstance(expression, ComponentTensor) and
+         not isinstance(expression.children[0], FlexiblyIndexed))):
```

Comment:
Update docstring? This is basically treating shaped things that are not `CT(FlexiblyIndexed)` as a `Variable`? (so no decomposition)?